File: Python Codes/3_RemoveElement.py
class Solution:
    def removeElement(self, nums, val):
        k = 0
        
        for i in range(len(nums)):
            if nums[i] != val:
                nums[k] = nums[i]
                k += 1
                
        return k

if __name__ == "__main__":
    n = int(input("Enter the number of elements: "))
    
    nums = []
    print("Enter the elements: ")
    for i in range(n):
        nums.append(int(input()))
    
    val = int(input("Enter the value to remove: "))
    
    solution = Solution()
    newLength = solution.removeElement(nums, val)
    
    print("Array after removal: ", end="")
    for i in range(newLength):
        print(nums[i], end=" ")
    print()
    
    print("New length:", newLength)

File: CPP Codes/3_Remove_Element.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution 
{
public:
    int removeElement(vector<int>& nums, int val) 
    {
        int k = 0;
        
        for (int i = 0; i < nums.size(); i++) 
        {
            if (nums[i] != val) 
            {
                nums[k] = nums[i];
                k++;
            }
        }
        
        return k;
    }
};

int main() 
{
    int n, val;
    
    cout << "Enter the number of elements: ";
    cin >> n;
    
    vector<int> nums(n);
    
    cout << "Enter the elements: ";
    for (int i = 0; i < n; i++) 
    {
        cin >> nums[i];
    }
    
    cout << "Enter the value to remove: ";
    cin >> val;
    
    Solution solution;
    int newLength = solution.removeElement(nums, val);
    
    cout << "Array after removal: ";
    for (int i = 0; i < newLength; i++) 
    {
        cout << nums[i] << " ";
    }
    cout << endl;
    
    cout << "New length: " << newLength << endl;
    
    return 0;
}